File: sol/1760/1760.py
class Solution:
    def minimumSize(self, nums: List[int], maxOperations: int) -> int:
        e = max(nums) + 1
        s = 1
        while s < e:
            m = (s + e) // 2
            if sum((x - 1) // m for x in nums) <= maxOperations:
                e = m
            else:
                s = m + 1
        return s

File: sol/1760/1760.cpp
struct Solution {
    int minimumSize(vector<int>& nums, int maxOperations) {
        int s = 1, e = 1e9 + 1;
        while (s < e){
            int m = (s + e) / 2, count = 0;
            for (int x : nums){
                count += (x - 1) / m;
            }
            if (count <= maxOperations){
                e = m;
            } else {
                s = m + 1;
            }
        }
        return s;
    }
};
